pallets/auctions/src/types/topup.rs

```diff
@@ -0,0 +1,143 @@
+use super::*;
+
+///
+/// Implementation of TopUpAuction
+///
+impl<T: Config> NftAuction<T::AccountId, T::AuctionId, BalanceOf<T>, Auction<T>, Bid<T>> for TopUpAuction<T> {
+	///
+	/// Creates a TopUp Auction
+	///
+	fn create(&self, sender: T::AccountId, auction: &Auction<T>) -> DispatchResult {
+		self.validate_data()?;
+		Pallet::<T>::validate_create(&self.common_data)?;
+		Pallet::<T>::handle_create(sender, auction, &self.common_data)?;
+
+		Ok(())
+	}
```

Comment:
AFAICT all three implementations are the same. You could think about providing a default implementation in the trait.